File: Powers Of 2/PowerOf2.cs
using System;

class Program  
{  
    static void Main(string[] args)  
    {  
    	 Console.WriteLine("The Nth term in the sequence is " + powerOf2()); 	        

        Console.ReadLine();
    }  

    public static int userInput() {
		while(true) {
			try {
				Console.WriteLine("Enter the value of n: ");
				int n = int.Parse(Console.ReadLine());
				return n;
			}
			catch(Exception e) {
				Console.WriteLine("Invalid input: Please enter an integer input.\n" + e);
			}
		}
	}

    public static int powerOf2() {
		int number;
		int n;
		
		while (true){
            
            try
            {
            	n = userInput();	// prompt the user to enter a number
            	
            	if (n > -1) 
                {
                    break;  // if an integer greater than -1 is inputed, we break the loop
                }
                Console.WriteLine("Invalid input, please enter a positive number or 0."); // if an integer less than zero is inputed, the loop will not be broken and this line will be printed
                
            }    
        
            catch(Exception e) // enter this catch block if a non integer is input
            {
            	Console.WriteLine("Invalid input, input must be an integer" + e);
            }          
        }	

		number = (int) Math.Pow(2, n);
        return number;
	} 
}

File: Powers Of 2/PowerOf2.py
import math

# Taking user input
def user_input():
    try:
        n = int(input("Enter n: "))
        return n
    except ValueError:  # This is needed to give line 13 an exception (Because it changes 'n')
        return user_input()


# Find required number in a power series
def find_power_of2():  # Power of 2 from wikipedia, means the n variable fills in the exponent
    n = user_input()

    while (
        True
    ):  # This loop is here just to prevent an error in case user inputs a negative number
        if n > -1:
            break
        else:
            print("Invalid input, please enter a positive number or 0.")
            n = user_input()

    number = math.pow(2, n)  # puts 2 to the power of n with function .pow
    return int(number)


# Main program
if __name__ == "__main__":
    print(f"The nth term in the sequence is {find_power_of2()}")
